### PR TITLE
Find the latest android.jar in the locally installed android studio

### PR DESCRIPTION
**Pull request by JCash**
## PR checklist

* [ ] Code
	* [ ] Add engine and/or editor unit tests.
	* [ ] New and changed code follows the overall code style of existing code
	* [ ] Add comments where needed
* [ ] Documentation
	* [ ] Make sure that API documentation is updated in code comments
	* [ ] Make sure that manuals are updated (in github.com/defold/doc)
* [ ] Prepare pull request and affected issue for automatic release notes generator
	* [ ] Pull request - Write a message that explains what this pull request does. What was the problem? How was it solved? What are the changes to APIs or the new APIs introduced? This message will be used in the generated release notes. Make sure it is well written and understandable for a user of Defold.
	* [ ] Pull request - Write a pull request title that in a sentence summarises what the pull request does. Do not include "Issue-1234 ..." in the title. This text will be used in the generated release notes.
	* [ ] Pull request - Link the pull request to the issue(s) it is closing. Use on of the [approved closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).
	* [ ] Affected issue - Assign the issue to a project. Do not assign the pull request to a project if there is an issue which the pull request closes.
	* [ ] Affected issue - Assign the "breaking change" label to the issue if introducing a breaking change.
	* [ ] Affected issue - Assign the "skip release notes" is the issue should not be included in the generated release notes.

----------

Example of a well written PR description:

1. Start with the user facing changes. This will end up in the release notes.
1. Add one of the GitHub approved closing keywords
1. Optionally also add the technical changes made. This is information that might help the reviewer. It will not show up in the release notes. Technical changes are identified by a line starting with one of these:
   1. `### Technical changes` 
   1. `Technical changes:`
   2. `Technical notes:`

```
There was a anomaly in the carbon chroniton propeller, introduced in version 8.10.2. This fix will make sure to reset the phaser collector on application startup.

Fixes #1234

### Technical changes
* Pay special attention to line 23 of phaser_collector.clj as it contains some interesting optimizations
* The propeller code was not taking into account a negative phase.
```

### CODE REVIEW

File: build_tools/sdk.py [hunk 1]
```diff
@@ -134,7 +134,7 @@ def _replace_pattern(s, patterns):
             s = s.replace(pattern, replace)
         return s
 
-    dirs.sort(key=lambda x: tuple(int(token) for token in _replace_pattern(x, replace_patterns).split('.')))
+    dirs.sort(key=lambda x: tuple(int(token) for token in _replace_pattern(x, replace_patterns).split('.')), reverse=True)
```
Comment (JCash) [inline build_tools/sdk.py:137]: Largest numbers first

File: build_tools/sdk.py [hunk 2]
```diff
@@ -260,7 +260,9 @@ def get_android_local_sdk_version(platform):
 
 def get_android_local_jar_path(verbose=False):
     sdkfolder = get_android_local_sdk_path()
-    path = os.path.join(sdkfolder, 'platforms', 'android-%s' % ANDROID_TARGET_API_LEVEL, 'android.jar')
+    platforms_folder = os.path.join(sdkfolder, 'platforms')
+    android_version = _get_latest_version_from_folders(platforms_folder, [('android-', '')])
+    path = os.path.join(sdkfolder, 'platforms', android_version, 'android.jar')
```
Comment (JCash) [inline build_tools/sdk.py:265]: Actually find the latest version in the sdk.

File: build_tools/sdk.py [hunk 3]
```diff
@@ -636,7 +638,7 @@ def _get_local_sdk_info(platform, verbose=False):
         info['version'] = get_android_local_sdk_version(platform)
         info['sdk'] = get_android_local_sdk_path(verbose)
         info['ndk'] = get_android_local_ndk_path(platform, verbose)
-        info['jar'] = get_android_local_jar_path(platform, verbose)
+        info['jar'] = get_android_local_jar_path(verbose)
```
Comment (JCash) [inline build_tools/sdk.py:641]: syntax error

File: build_tools/sdk.py (hunk 1 above)
Comment (britzl) [reply build_tools/sdk.py:137]: Side note and unrelated, but I really don't like python lambda functions. I really have to pause and look closely at the code to understand what it really does.... :-/

**Review by britzl — APPROVED** [review]

**Pull request merged**